### PR TITLE
Improve CPU binding

### PR DESCRIPTION
**Pull request by yuyichao**
* Allow specifying a particular CPU to bind to
* Take into account current cpuset CGroup when picking a random CPU
* Fix document about CPU binding

### CODE REVIEW

File: src/RecordSession.cc [hunk 1]
```diff
@@ -95,12 +153,19 @@ static int get_num_cpus() {
...
   if (bind_cpu == RecordSession::UNBOUND_CPU) {
     return -1;
   }
+  else if (bind_cpu >= 0) {
+    return bind_cpu;
+  }
+
+  int cpu = get_random_cpu_cgroup();
+  if (cpu >= 0)
+    return cpu;
```
Comment (rocallahan) [inline src/RecordSession.cc]: {} around if body

File: src/RecordSession.cc [hunk 2]
```diff
@@ -95,12 +153,19 @@ static int get_num_cpus() {
  * Pick a CPU at random to bind to, unless --cpu-unbound has been given,
  * in which case we return -1.
  */
-static int choose_cpu(RecordSession::BindCPU bind_cpu) {
+static int choose_cpu(int bind_cpu) {
   if (bind_cpu == RecordSession::UNBOUND_CPU) {
     return -1;
   }
+  else if (bind_cpu >= 0) {
```
Comment (rocallahan) [inline src/RecordSession.cc]: Just drop the `else`. Also, move the comment explaining why we're binding to a CPU up to here.

**Comment by yuyichao** [comment]
Done

**Comment by rocallahan** [comment]
Thanks!

**Pull request merged**